psi4/broker.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2013-2018 Thomas Spura <thomas.spura@gmail.com>
+from __future__ import print_function
+
+import string
+import sys
+import time
+import numpy as np
+from numpy.testing import assert_array_almost_equal as assert_equal
+
+import psi4
+try:
+    from ipi.interfaces.sockets import Client
+    ipi_available = True
+except ImportError:
+    ipi_available = False
+    # Define Client to enable testing of the Broker in the unittests
+    class Client(object): pass
+
+
+class Broker(Client):
+    def __init__(self, options, genbas=None, serverdata=False):
+        self.serverdata = serverdata
+        if not ipi_available:
+            print("i-pi is not available for import:",
+                  "The broker infrastructure will not be available!")
+            super(Broker, self).__init__()
+        elif serverdata:
+            mode, address, port = serverdata.split(":")
+            mode = string.lower(mode)
+            super(Broker, self).__init__(address=address, port=port,
+                                      mode=mode)
+        else:
+            super(Broker, self).__init__(_socket=False)
+        self.options = options
+
+        print("PSI4 options:")
+        for item, value in self.options.items():
+            print(item, value)
+            if item not in ["LOT", "multiplicity", "charge"]:
+                psi4.core.set_global_option(item, value)
+        psi4.core.IO.set_default_namespace("xwrapper")
+
+        self.timing = {}
+
+    def run_frc(self, pos):
+        """Fetch force, energy of PSI.
+        """
+        if len(pos.shape) == 1:
+            pos = pos.reshape((len(pos) / 3, 3))
```

Comment:
Probably should use floor division here `len(pos) // 3` to prevent float oddness. Or even better `pos.reshape(-1, 3)`.